### PR TITLE
add image pull retries

### PR DESCRIPTION
**Pull request by BenTheElder**
fixes https://github.com/kubernetes-sigs/kind/issues/29

TODO: make log level configurable and cleanup logging ref https://github.com/kubernetes-sigs/kind/issues/23

### CODE REVIEW

File: pkg/docker/docker.go [hunk 1]
```diff
@@ -18,17 +18,42 @@ limitations under the License.
...
+			log.Infof("Trying again to pull image: %s ...", image)
+			// TODO(bentheelder): add some backoff / sleep?
+			err = exec.Command("docker", "pull", image).Run()
+			if err == nil {
+				break
+			}
+		}
+	}
+	if err != nil {
+		log.Infof("Failed to pull image: %s", image)
```
Comment (krzyzacy) [inline pkg/docker/docker.go]: nit: log error?

Comment (BenTheElder) [reply pkg/docker/docker.go]: done

**Comment by BenTheElder** [comment]
(resolved review, re-lgtm so we can soak this, image pulling is one of the major remaining flakes..)

File: pkg/docker/docker.go (hunk 1 above)
Comment (BenTheElder) [reply pkg/docker/docker.go]: thanks :-)

**Pull request merged**